### PR TITLE
Micro-optimization: only split `acct` into two Strings

### PR DESCRIPTION
**Pull request by postmodern**
Since `acct` is split by `@` and assigned to `username` and `domain`, we only need to split `acct` into two Strings.

### CODE REVIEW

File: app/models/account_alias.rb [hunk 1]
```diff
@@ -29,7 +29,7 @@ def acct=(val)
   end
 
   def pretty_acct
-    username, domain = acct.split('@')
+    username, domain = acct.split('@',2)
```
Comment (sunny) [inline app/models/account_alias.rb:32]:
```suggestion
    username, domain = acct.split('@', 2)
```

Comment (postmodern) [reply app/models/account_alias.rb]: Manually edited and force pushed.

**Comment by ClearlyClaire** [comment]
Does that actually bring any benefit?

**Comment by postmodern** [comment]
> Does that actually bring any benefit?

Yes, by about 50%.

```ruby
#!/usr/bin/env ruby
require 'benchmark'

Benchmark.bm do |b|
  n = 1_000_000

  account = 'foo@bar@baz@qux.com'

  b.report('String#split(\'@\')') do
    n.times do
      user, domain = account.split('@')
    end
  end

  b.report('String#split(\'@\',2)') do
    n.times do
      user, domain = account.split('@',2)
    end
  end
end
```

```
       user     system      total        real      
String#split('@')  0.671949   0.002190   0.674139 (  0.679637)
String#split('@',2)  0.436089   0.000990   0.437079 (  0.439453)
```

**Comment by ClearlyClaire** [comment]
In this case, however, it is known to have at most one `@` because that method is only ever used on accts that are saved to the database and deemed valid as far as I can tell.

**Comment by postmodern** [comment]
> In this case, however, it is known to have at most one `@` because that method is only ever used on accts that are saved to the database and deemed valid as far as I can tell.

Are you sure that is always going to be the case? Better to be explicit since in this case it is extracting only two components.

**Review by ClearlyClaire — APPROVED** [review]

**Comment by ClearlyClaire** [comment]
I'm confident that won't change anytime soon, and it's far from being a hot spot in the code path. But I see no downside to this either way.

**Pull request merged**